src/core/src/service_interfaces/LifecycleManagerARC.py

```diff
@@ -0,0 +1,187 @@
+# Copyright 2020 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.7+
+
+import json
+import os
+import shutil
+import time
+from core.src.bootstrap.Constants import Constants
+from core.src.service_interfaces.LifecycleManager import LifecycleManager
+
+class LifecycleManagerArc(LifecycleManager):
+    """Class for managing the core code's lifecycle within the extension wrapper"""
+
+    def __init__(self, env_layer, execution_config, composite_logger, telemetry_writer):
+        super(LifecycleManagerArc,self).__init__(env_layer,execution_config,composite_logger,telemetry_writer)
+
+        # Handshake file paths
+        self.ext_state_file_path = os.path.join(self.execution_config.config_folder, Constants.EXT_STATE_FILE)
+        self.core_state_file_path = os.path.join(self.execution_config.config_folder, Constants.CORE_STATE_FILE)
+        #Writing to log
+        self.composite_logger.log_debug("Initializing LifecycleManagerArc")
+        #Variables
+        self.arc_root = "/var/lib/waagent/"
+        self.core_file_name = "CoreState.json"
```

Comment:
Is this for Arc only? Just checking if variable can be disambiguated (prefix with arc_ if necessary) -- need some more context. Many close this comment after reviewing more.